src/modules/Elsa.Dapper/Services/Store.cs

```diff
@@ -245,6 +245,11 @@ public async Task<long> DeleteAsync(Action<ParameterizedQuery> filter, Cancellat
     {
         var query = _dbConnectionProvider.CreateQuery().Delete(TableName);
         filter(query);
+        
+        // If there are no conditions, we don't want to delete all records.
+        if (!query.Parameters.ParameterNames.Any())
```

Comment:
Do we want to support a delete all statement?

Reply:
Probably eventually, yes, but I couldn't think of a decent solution on the spot, and I don't think we have a current use case for it that warrants additional work on the Dapper provider.